web/middlewares/permissions.go

```diff
@@ -327,6 +327,21 @@ func AllowVFS(c echo.Context, v permission.Verb, o vfs.Fetcher) error {
 	return nil
 }
 
+// CanWriteToAnyDirectory checks that the context permission allows to write to
+// a directory on the VFS.
+func CanWriteToAnyDirectory(c echo.Context) error {
+	pdoc, err := GetPermission(c)
+	if err != nil {
+		return err
+	}
+	for _, rule := range pdoc.Permissions {
+		if rule.Type == consts.Files && rule.Verbs.Contains(permission.POST) {
+			return nil
+		}
+	}
+	return ErrForbidden
+}
```

Comment:
Thanks to that code, I just read https://gist.github.com/posener/a303becac35835ad7bf5e15fe061893e 👍